db/R/query.format.vars.R

```diff
@@ -34,53 +34,63 @@ query.format.vars <- function(input.id,con,format.id){
+    vars$variable_id <- as.numeric(fv$variable_id)
   
   
-  # Fill in CF vars
-  # This will ultimately be useful when looking at met variables where CF != Bety
-  # met <- read.csv(system.file("/data/met.lookup.csv", package= "PEcAn.data.atmosphere"), header = T, stringsAsFactors=FALSE)
+    # Fill in CF vars
+    # This will ultimately be useful when looking at met variables where CF != Bety
+    # met <- read.csv(system.file("/data/met.lookup.csv", package= "PEcAn.data.atmosphere"), header = T, stringsAsFactors=FALSE)
   
-  #Fill in MstMIP vars 
-  #All pecan output is in MstMIP variables 
+    #Fill in MstMIP vars 
+    #All pecan output is in MstMIP variables 
   
-  bety_mstmip <- read.csv(system.file("bety_mstmip_lookup.csv", package= "PEcAn.DB"), header = T, stringsAsFactors=FALSE)
+    bety_mstmip <- read.csv(system.file("bety_mstmip_lookup.csv", package= "PEcAn.DB"), header = T, stringsAsFactors=FALSE)
   
-  vars_full <- merge(fv, merge(vars, bety_mstmip, by = "bety_name", all.x = TRUE), by="variable_id", all=TRUE) # not sure if all=TRUE is appropriate here or not.
+    vars_full <- merge(fv, merge(vars, bety_mstmip, by = "bety_name", all.x = TRUE), by="variable_id", all=TRUE) # not sure if all=TRUE is appropriate here or not.
   
-  vars_full$pecan_name <- vars_full$mstmip_name
-  vars_full$pecan_units <- vars_full$mstmip_units
-  ind <- is.na(vars_full$pecan_name)
-  vars_full$pecan_name[ind] <- vars_full$bety_name[ind]
-  vars_full$pecan_units[ind] <- vars_full$bety_units[ind]
+    vars_full$pecan_name <- vars_full$mstmip_name
+    vars_full$pecan_units <- vars_full$mstmip_units
+    ind <- is.na(vars_full$pecan_name)
+    vars_full$pecan_name[ind] <- vars_full$bety_name[ind]
+    vars_full$pecan_units[ind] <- vars_full$bety_units[ind]
   
-  header <- as.numeric(f$header)
-  skip <- ifelse(is.na(as.numeric(f$skip)),0,as.numeric(f$skip))
+    header <- as.numeric(f$header)
+    skip <- ifelse(is.na(as.numeric(f$skip)),0,as.numeric(f$skip))
   
-  # Final format list
-  format <- list(file_name = f$name,
-                 mimetype = f$mimetype,
-                 vars = vars_full,
-                 skip = skip, 
-                 header = header,
-                 na.strings=c("-9999","-6999","9999"), # This shouldn't be hardcoded in, but not specified in format table ?
-                 site = site.id,
-                 lat = site.lat,
-                 lon = site.lon
-                 )
+    # Final format list
+    format <- list(file_name = f$name,
+                   mimetype = f$mimetype,
+                   vars = vars_full,
+                   skip = skip, 
+                   header = header,
+                   na.strings=c("-9999","-6999","9999"), # This shouldn't be hardcoded in, but not specified in format table ?
+                   site = site.id,
+                   lat = site.lat,
+                   lon = site.lon
+                   )
+  } else {
+    format <- list(file_name = f$name,
```

Comment:
If I understand this correctly, this bug fix will return a format object without any variables if there are no variables associated with the format. But that begs the question, what use is a file that has no variables in it?  Would it not be better to throw an error message telling the user which Formats record need fixing?  I worry that enabling this to propagate farther down stream will just make it harder for the end user to debug